RedPanda + kafka-streams stop working in upstream

### Describe the bug

Related to commit: https://github.com/quarkusio/quarkus/commit/f9f8b9236683d21bc5fe39f505880ca46e96a133

Quarkus version: upstream
Extension: quarkus-kafka-streams

Kafka devService (Redpanda) is throwing the following error when is trying to consume a topic through Kafka streams

```
10:06:11,786 INFO  [app] Written offsets would not be recorded and no more records would be sent since this is a fatal error.: org.apache.kafka.common.errors.UnknownServerException: The server experienced an unexpected error when processing the request.
10:06:11,787 INFO  [app] 10:06:09,219 stream-thread [login-denied-aggregator-b69141f2-360b-414e-8877-694c6a43b22a-StreamThread-1] task [0_0] Failed to flush cache of store login-aggregation-store: : org.apache.kafka.streams.errors.StreamsException: Error encountered sending record to topic login-denied-aggregator-login-aggregation-store-changelog for task 0_0 due to:
10:06:11,788 INFO  [app] org.apache.kafka.common.errors.UnknownServerException: The server experienced an unexpected error when processing the request.
10:06:11,788 INFO  [app] Written offsets would not be recorded and no more records would be sent since this is a fatal error.
10:06:11,789 INFO  [app]        at org.apache.kafka.streams.processor.internals.RecordCollectorImpl.recordSendError(RecordCollectorImpl.java:209)
10:06:11,790 INFO  [app]        at org.apache.kafka.streams.processor.internals.RecordCollectorImpl.lambda$send$0(RecordCollectorImpl.java:196)
10:06:11,791 INFO  [app]        at org.apache.kafka.clients.producer.KafkaProducer$InterceptorCallback.onCompletion(KafkaProducer.java:1418)
10:06:11,792 INFO  [app]        at org.apache.kafka.clients.producer.internals.ProducerBatch.completeFutureAndFireCallbacks(ProducerBatch.java:273)
10:06:11,793 INFO  [app]        at org.apache.kafka.clients.producer.internals.ProducerBatch.done(ProducerBatch.java:234)
10:06:11,794 INFO  [app]        at org.apache.kafka.clients.producer.internals.ProducerBatch.completeExceptionally(ProducerBatch.java:198)
10:06:11,794 INFO  [app]        at org.apache.kafka.clients.producer.internals.Sender.failBatch(Sender.java:758)
10:06:11,795 INFO  [app]        at org.apache.kafka.clients.producer.internals.Sender.failBatch(Sender.java:743)
10:06:11,796 INFO  [app]        at org.apache.kafka.clients.producer.internals.Sender.failBatch(Sender.java:695)
10:06:11,796 INFO  [app]        at org.apache.kafka.clients.producer.internals.Sender.completeBatch(Sender.java:634)
10:06:11,797 INFO  [app]        at org.apache.kafka.clients.producer.internals.Sender.lambda$null$1(Sender.java:575)
10:06:11,797 INFO  [app]        at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
10:06:11,798 INFO  [app]        at org.apache.kafka.clients.producer.internals.Sender.lambda$handleProduceResponse$2(Sender.java:562)
10:06:11,798 INFO  [app]        at java.base/java.lang.Iterable.forEach(Iterable.java:75)
10:06:11,799 INFO  [app]        at org.apache.kafka.clients.producer.internals.Sender.handleProduceResponse(Sender.java:562)
10:06:11,800 INFO  [app]        at org.apache.kafka.clients.producer.internals.Sender.lambda$sendProduceRequest$5(Sender.java:836)
10:06:11,800 INFO  [app]        at org.apache.kafka.clients.ClientResponse.onComplete(ClientResponse.java:109)
10:06:11,801 INFO  [app]        at org.apache.kafka.clients.NetworkClient.completeResponses(NetworkClient.java:583)
10:06:11,801 INFO  [app]        at org.apache.kafka.clients.NetworkClient.poll(NetworkClient.java:575)
10:06:11,801 INFO  [app]        at org.apache.kafka.clients.producer.internals.Sender.runOnce(Sender.java:328)
10:06:11,802 INFO  [app]        at org.apache.kafka.clients.producer.internals.Sender.run(Sender.java:243)
10:06:11,802 INFO  [app]        at java.base/java.lang.Thread.run(Thread.java:829)
10:06:11,802 INFO  [app] Caused by: org.apache.kafka.common.errors.UnknownServerException: The server experienced an unexpected error when processing the request.

```

Could be related to Kafka 3.2.1 


### Expected behavior

Events are consumed from a Kafka stream consumer

### Actual behavior

Events are not consumed from a Kafka stream consumer

### How to Reproduce?

`git clone git@github.com:quarkus-qe/quarkus-test-suite.git`

`mvn clean verify -Dall-modules -pl messaging/kafka-streams-reactive-messaging -Dit.test=DevModeKafkaStreamIT#testAlertMonitorEventStream`

Note: be sure that `testAlertMonitorEventStream` that is located on class `BaseKafkaStreamTest` is not disabled

### Output of `uname -a` or `ver`

Linux localhost.localdomain 5.18.13-100.fc35.x86_64 #1 SMP PREEMPT_DYNAMIC Fri Jul 22 14:20:24 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk version "11.0.16" 2022-07-19 OpenJDK Runtime Environment Temurin-11.0.16+8 (build 11.0.16+8) OpenJDK 64-Bit Server VM Temurin-11.0.16+8 (build 11.0.16+8, mixed mode)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

upstream

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.3

### Additional information

_No response_

--- a/extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaDevServicesBuildTimeConfig.java
+++ b/extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaDevServicesBuildTimeConfig.java
@@ -45,7 +45,7 @@ public class KafkaDevServicesBuildTimeConfig {
      * See https://github.com/strimzi/test-container and https://quay.io/repository/strimzi-test-container/test-container
      * <p>
      */
-    @ConfigItem(defaultValue = "docker.io/vectorized/redpanda:v21.11.3")
+    @ConfigItem(defaultValue = "docker.io/vectorized/redpanda:v22.1.7")
     public String imageName;
 
     /**

--- a/extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/RedPandaKafkaContainer.java
+++ b/extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/RedPandaKafkaContainer.java
@@ -57,6 +57,7 @@ final class RedPandaKafkaContainer extends GenericContainer<RedPandaKafkaContain
         command += "--memory 1G --overprovisioned --reserve-memory 0M ";
         command += String.format("--kafka-addr %s ", getKafkaAddresses());
         command += String.format("--advertise-kafka-addr %s ", getKafkaAdvertisedAddresses());
+        command += "--set redpanda.auto_create_topics_enabled=true ";
         if (redpandaConfig.transactionEnabled) {
             command += "--set redpanda.enable_idempotence=true ";
             command += "--set redpanda.enable_transactions=true ";

